uaclient/tests/test_ua_update_messaging.py

```diff
@@ -0,0 +1,288 @@
+import datetime
+import mock
+import os
+
+import pytest
+
+from uaclient.defaults import (
+    BASE_UA_URL,
+    CONTRACT_EXPIRY_PENDING_DAYS,
+    CONTRACT_EXPIRY_GRACE_PERIOD_DAYS,
+)
+from uaclient.status import (
+    MESSAGE_CONTRACT_EXPIRED_GRACE_PERIOD_TMPL,
+    MESSAGE_CONTRACT_EXPIRED_MOTD_TMPL,
+    MESSAGE_CONTRACT_EXPIRED_APT_TMPL,
+    MESSAGE_CONTRACT_NEAR_EXPIRED_TMPL,
+    MESSAGE_ANNOUNCE_ESM,
+    MESSAGE_UBUNTU_NO_WARRANTY,
+)
+from uaclient import util
+
+from lib.ua_update_messaging import (
+    format_contract_expiry_messages,
+    format_esm_announcement_message,
+    update_apt_and_motd_messages,
+)
+
+M_MESSAGE_PATH = "lib.ua_update_messaging."
```

Comment:
Personal preference would be to hard-code this string in all the `mock.patch` calls so I don't have to remember what the variable is when figuring out what is mocked, but :shrug: nbd either way